colibri/src/globaldb.rs

```diff
@@ -62,6 +63,28 @@ impl GlobalDB {
                 })
         })
     }
+
+    /// Get all active RPC endpoints for a specific blockchain.
+    pub async fn get_rpc_nodes(&self, blockchain: SupportedBlockchain) -> Result<Vec<Node>> {
+        let conn = self.conn.lock().await;
```

Comment:
why return Result rather than the vec itself?

Reply:
unless i'm wrong but isn't `Result<>` or `Option<>` the way rust handles return values or am i missing something?

Reply:
Yes but you can also return an empty vec. It is not mandatory to wrap everything on result/option